hazelcast-spring/src/test/java/com/hazelcast/spring/transaction/DummyObject.java

```diff
@@ -35,15 +32,27 @@ public Long getId() {
         return id;
     }
 
-    public void setId(Long id) {
-        this.id = id;
-    }
-
     public String getString() {
         return string;
     }
 
-    public void setString(String string) {
-        this.string = string;
+    @Override
+    public int hashCode() {
+        if (id == null) {
+            return super.hashCode();
+        }
+
+        return id.hashCode();
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (id == null || !(obj instanceof DummyObject)) {
```

Comment:
What if both object have id==null? Then they are not equal?

Reply:
yeah, this is not like `null==null` but initially I assumed that objects with `null` ids shouldn't be equal. Anyway, I can fix it if you think it should.

Reply:
Yes, it would be fine!

Reply:
done